src/datasets/user-white-noise.dataset.ts

```diff
@@ -0,0 +1,95 @@
+import { Dataset } from './types';
+import {
+  activeUserMeasure,
+  browserCategory,
+  countryCategory,
+  eventCountMeasure,
+  revenueMeasure,
+  sourceCategory,
+} from '../models/data-cube/presets';
+import { generateCube } from '../models/data-cube/generation';
+import { humanizeMeasureName } from '../utils/formatters';
+import { createTimeSeriesQuery } from './queries/time-series.query';
+import { createLineChartMeta } from './metas/line-chart.meta';
+import { createTabbedChartsMeta } from './metas/tabbed-charts.meta';
+import { PreferenceMeta } from '../services/preference/types';
+import { createDefault } from '../utils/preferences';
+
+export interface Config {
+  avgHits: number;
+  hitStdDev: number;
+  avgUsers: number;
+  userStdDev: number;
+  avgSessionsPerUser: number;
+  sessionsPerUserStdDev: number;
+}
+
+export const configMeta: PreferenceMeta<Config> = {
+  avgHits: {
+    type: 'number',
+    defaultValue: 10000,
+  },
+  hitStdDev: {
+    type: 'number',
+    defaultValue: 100,
+  },
+  avgUsers: {
+    type: 'number',
+    defaultValue: 100,
+  },
+  userStdDev: {
+    type: 'number',
+    defaultValue: 1,
+  },
+  avgSessionsPerUser: {
+    type: 'number',
+    defaultValue: 5,
+  },
+  sessionsPerUserStdDev: {
+    type: 'number',
+    defaultValue: 3,
+  },
+};
+
+export function create(config: Config): Dataset {

```

Comment:
You don't need to do this in this PR, but generally it's nice to have a small documentation for exported methods, services or any significant chunk of code so that someone can get the gist of it without reading through all the code

Reply:
It raises another design question here. Actually `Config`, `configMeta`, and `create` are three necessary exports/properties for all dataset module/object, it is nice to have a abstract parent interface or class for all datasets to extend. However, they would be all static and immutable properties and I don't want developers to create instance of dataset if I make it a class, so I make them modules when adding dataset support.
The bad things of this design are that it is hard to build mock datasets for testing and do the type inference on the map of dataset modules:
https://github.com/googleinterns/guide-doge/blob/0b203248399232462673de30cdcddd2e72608dd9/src/datasets/index.ts#L4

I would like to refactor it if you have any good ideas.

Reply:
Off-topic but it would be great if TypeScript supports module extending interface.

Something like `declare module extends AbstractDataset;` and all the members of `AbstractDataset` has to be implemented and exported, so people don't need to have an extra wrapper to make it type-safe.